provider/kubernetes/kubernetes_test.go

```diff
@@ -2262,6 +2263,297 @@ func TestBasicAuthInTemplate(t *testing.T) {
+					"example.org": {
+						Rule: "Host:example.org",
+					},
+				},
+			},
+		},
+		TLSConfiguration: []*tls.Configuration{
+			{
+				EntryPoints: []string{"example.com"},
+				Certificate: &tls.Certificate{
+					CertFile: "-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----",
+					KeyFile:  "-----BEGIN PRIVATE KEY-----\n-----END PRIVATE KEY-----",
+				},
+			},
+		},
+	}
+
+	assert.Equal(t, expected, actual)
+}
+
+func TestGetTLSConfigurations(t *testing.T) {
+	testIngress := v1beta1.Ingress{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "test-ingress",
+			Namespace: "testing",
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: "ep1.example.com",
+				},
+				{
+					Host: "ep2.example.com",
+				},
+			},
+			TLS: []v1beta1.IngressTLS{{SecretName: "test-secret"}},
+		},
+	}
+
+	tests := []struct {
+		desc      string
+		ingress   *v1beta1.Ingress
+		client    Client
+		result    []*tls.Configuration
+		errResult string
+	}{
+		{
+			desc:    "error_api_client",
+			ingress: &testIngress,
+			client: clientMock{
+				apiSecretError: errors.New("api secret error"),
+			},
+			errResult: "secret testing/test-secret does not exist: api secret error",
+		},
+		{
+			desc:      "secret_not_found",
+			ingress:   &testIngress,
+			client:    clientMock{},
+			errResult: "secret testing/test-secret does not exist",
+		},
+		{
+			desc:    "secret_crt_missing",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.key": []byte("tls-key"),
+						},
+					},
+				},
+			},
+			errResult: "secret testing/test-secret must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.crt'",
+		},
+		{
+			desc:    "secret_key_missing",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.crt": []byte("tls-crt"),
+						},
+					},
+				},
+			},
+			errResult: "secret testing/test-secret must have two entries named 'tls.crt' and 'tls.key': missing entry 'tls.key'",
+		},
+		{
+			desc:    "default_wildcard_behavior",
+			ingress: &testIngress,
+			client: clientMock{
+				secrets: []*v1.Secret{
+					{
+						ObjectMeta: v1.ObjectMeta{
+							Name:      "test-secret",
+							Namespace: "testing",
+						},
+						Data: map[string][]byte{
+							"tls.crt": []byte("tls-crt"),
+							"tls.key": []byte("tls-key"),
+						},
+					},
+				},
+			},
+			result: []*tls.Configuration{
+				{
+					Certificate: &tls.Certificate{
+						CertFile: tls.FileOrContent("tls-crt"),
+						KeyFile:  tls.FileOrContent("tls-key"),
+					},
+					EntryPoints: []string{
+						"ep1.example.com",
+						"ep2.example.com",
+					},
+				},
+			},
+		},
```

Comment:
Let's add another test case for the non-wildcard scenario. I know it's already covered by `TestTLSSecretLoad`, but in that test it could potentially be any other code path that leads to a valid TLS configuration we can compare against.

Including the dedicated test case in `TestGetTLSConfigurations` makes sure all code paths of the `getTLSConfigurations` function are covered by the owning test function while giving us the freedom to change `TestTLSSecretLoad`  in the future.